runtimes/pythonrt/pythonrt_test.go

```diff
@@ -253,3 +264,51 @@ func TestCleanup(t *testing.T) {
 	_, err = svc.initialCall(context.Background(), "greet", nil) // will panic
 	require.Error(t, err)
 }
+
+var progErrCode = []byte(`
+def handle(event):
+    1 / 0
+`)
+
+func TestProgramError(t *testing.T) {
+	skipIfNoPython(t)
+
+	pyFile := "progerr.py"
+	var buf bytes.Buffer
+	tw := tar.NewWriter(&buf)
+	hdr := &tar.Header{
+		Name: pyFile,
+		Size: int64(len(progErrCode)),
+		Mode: 0644,
+	}
+	err := tw.WriteHeader(hdr)
+	require.NoError(t, err)
+	_, err = tw.Write(progErrCode)
+	tw.Close()
+
+	compiled := map[string][]byte{
+		archiveKey: buf.Bytes(),
+	}
+
+	require.NoError(t, err)
+	svc := newSVC(t)
+
+	runID := sdktypes.NewRunID()
+	mod, values := newValues(t, runID)
+	cbs := newCallbacks(svc)
+	ctx, cancel := testCtx(t)
+	defer cancel()
+	_, err = svc.Run(ctx, runID, pyFile, compiled, values, cbs)
+	require.NoError(t, err)
+
+	xid := sdktypes.NewExecutorID(runID)
+	fn, err := sdktypes.NewFunctionValue(xid, "handle", nil, nil, mod)
+	require.NoError(t, err, "new function")
+
+	kwargs := map[string]sdktypes.Value{}
+
+	_, err = svc.Call(ctx, fn, nil, kwargs)
+	require.Error(t, err)
+	// There no way to check that err is a ProgramError since it's wrapped by unexported programError
+	require.Contains(t, err.Error(), "division by zero")
```

Comment:
could we check that it contains traceback/function/line?

Reply:
Hmm, the stack trace is:
```
division by zero
         [0] <frozen runpy>:198.1 _run_module_as_main
         [1] <frozen runpy>:88.1 _run_code
         [2] /tmp/ak-2638291852/ak_runner/__main__.py:204.1 <module>
         [3] /tmp/ak-2638291852/ak_runner/__main__.py:135.1 run
```

I guess we want the user code there as well, checking...

Reply:
Thanks @ig-ra, now the traceback is fill.